SimMuon/GEMDigitizer/src/ME0ReDigiProducer.cc

```diff
@@ -19,259 +19,385 @@
-	newR = CLHEP::RandGaussQ::shoot(engine, oldR, radialResolution_);
-      
-      // calculate the new position in local coordinates
-      const GlobalPoint radialSmearedGP(GlobalPoint::Cylindrical(newR, oldGP.phi(), oldGP.z()));
-      LocalPoint radialSmearedLP = roll->toLocal(radialSmearedGP);
-      
-      // new y position after smearing
-      const float targetYResolution(sqrt(newYResolution_*newYResolution_ - oldYResolution_ * oldYResolution_));
-      float newLPy = radialSmearedLP.y();
-      if (me0Digi.prompt())
-	newLPy = CLHEP::RandGaussQ::shoot(engine, radialSmearedLP.y(), targetYResolution);
-      
-      const ME0EtaPartition* newPart = roll;
-      LocalPoint newLP(radialSmearedLP.x(), newLPy, radialSmearedLP.z());
-      GlobalPoint newGP(newPart->toGlobal(newLP));
-      	
-      // check if digi moves one up or down roll
-      int newRoll = detId.roll();
-      if (newLP.y() > height)  --newRoll;
-      if (newLP.y() < -height) ++newRoll;
-
-      if (newRoll != detId.roll()){
-	// check if new roll is possible
-	if (newRoll < ME0DetId::minRollId || newRoll > ME0DetId::maxRollId){
-	  newRoll = detId.roll();
+
+	LogDebug("ME0ReDigiProducer::buildDigis") << "Begin bulding digis."<<std::endl;
+	ME0DigiPreRecoCollection::DigiRangeIterator me0dgIt;
+	for (me0dgIt = input_digis.begin(); me0dgIt != input_digis.end();
+			++me0dgIt){
+
+		const auto& me0Id = (*me0dgIt).first;
+		LogTrace("ME0ReDigiProducer::buildDigis") << "Starting with roll: "<< me0Id<<std::endl;
+
+		//setup map for this chamber/eta partition
+		ChamberDigiMap chDigiMap;
+
+		int newDigiIdx = 0;
+		const ME0DigiPreRecoCollection::Range& range = (*me0dgIt).second;
+		for (ME0DigiPreRecoCollection::const_iterator digi = range.first;
+				digi != range.second;digi++) {
+			LogTrace("ME0ReDigiProducer::buildDigis") << std::endl<< "(" <<digi->x() <<","<< digi->y()<<","<<digi->tof()<<","<<digi->pdgid()<<","<<digi->prompt()<<")-> ";
+
+			//If we don't readout this layer skip
+			if(!layerReadout[me0Id.layer() -1 ]) {
+				output_digimap.insertDigi(me0Id, -1);
+				continue;
+			}
+
+			//if neutron and we are filtering skip
+			if(!digi->prompt() && neutronAcceptance < 1.0 )
+				if (CLHEP::RandFlat::shoot(engine) > neutronAcceptance){
+					output_digimap.insertDigi(me0Id, -1);
+					continue;
+				}
+
+			//smear TOF if necessary
+			float tof = digi->tof() + (timeResolution < 0 ? 0.0 : CLHEP::RandGaussQ::shoot(engine, 0, timeResolution));
+
+			//Values used to fill objet
+			int mapPartIDX = me0Id.roll() -1;
+			int strip = 0;
+			LocalPoint  digiLocalPoint;
+			LocalError  digiLocalError;
+			if(useBuiltinGeo){
+				getStripProperties(geometry->etaPartition(me0Id),&*digi,tof,strip,digiLocalPoint,digiLocalError);
+			} else {
+				mapPartIDX = getCustomStripProperties(me0Id,&*digi,tof,strip,digiLocalPoint,digiLocalError);
+
+			}
+
+			//filter if outside of readout window
+			const int bxIdx = std::round(tof/25.0);
```

Comment:
Ideally "25" would be a constant member of the class (here and elsewhere, e.g. a few lines below)